machine_learning/loss_functions/huber_loss.py

```diff
@@ -0,0 +1,52 @@
+"""
+Huber Loss Function
+
+Description:
+Huber loss function describes the penalty incurred by an estimation procedure.
+It serves as a measure of the model's accuracy in regression tasks.
+
+Formula:
+Huber Loss = if |y_true - y_pred| <= delta then 0.5 * (y_true - y_pred)^2
+             else delta * |y_true - y_pred| - 0.5 * delta^2
+
+Source:
+[Wikipedia - Huber Loss](https://en.wikipedia.org/wiki/Huber_loss)
+"""
+
+import numpy as np
+
+
+def huber_loss(y_true: np.ndarray, y_pred: np.ndarray, delta: float) -> float:
+    """
+    Calculate the Huber Loss.
+
+    Parameters:
+    - y_true: The true values (ground truth).
+    - y_pred: The predicted values.
+
+    Returns:
+    - huber_loss: The mean of Huber Loss between y_true and y_pred.
+
+    Example usage:
+    >>> true_values = np.array([0.9, 10.0, 2.0, 1.0, 5.2])
+    >>> predicted_values = np.array([0.8, 2.1, 2.9, 4.2, 5.2])
+    >>> huber_loss(true_values, predicted_values, 1.0)
+    2.1020000000000003
```

Comment:
The true Huber loss value is 2.102. The trailing 3 is due to floating point errors. Please independently verify your tests' expected outputs, and please use the `np.isclose()` function in your tests in order to handle these floating point errors.

Reply:
I have added np.isclose() to the doctests to handle the floating point errors